### PR TITLE
Reject invalid --expected-shred-version

### PR DESCRIPTION
**Pull request by ryoqun**
#### Problem

The value given to `--expected-shred-version` isn't validated at all. Any string value can be given like: `--expected-shread-version skip_any_corrupted_record` and validator works and the option is just ignored.
This confuses users and makes debugging/investigation/support hard.

This `ok()` causes this bug, squashing any parsing errors into `None`, which is same as if the option isn't given: https://github.com/solana-labs/solana/blob/593ad809541336da5790bfed9d5899a3bcc08e8f/validator/src/main.rs#L1426

#### Summary of Changes

Add appropriate clap `.validator()`.

BEFORE:

```
$ ps auxf
...
solana-validator ... --expected-shred-version 2323442344

$ ps auxf
...
solana-validator ... --expected-shred-version baddd
```

AFTER:

```
error: Invalid value for '--expected-shred-version <VERSION>': error parsing 'jajaj': invalid digit found in string
error: Invalid value for '--expected-shred-version <VERSION>': error parsing '2323442344': number too large to fit in target type
```

In the strictest sense, this breaks compatibility but I think this could sneak in v1.4 and v1.5, assuming we add appropriate small note in the next patch release note?

Fixes #

### CODE REVIEW

**Comment by t-nelson** [comment]
> In the strictest sense, this breaks compatibility but I think this could sneak in v1.4 and v1.5, assuming we add appropriate small note in the next patch release note?

I think this is fine. We'll only be breaking validators who are misusing it now and would likely be better off learning this fact

File: validator/src/main.rs [hunk 1]
```diff
@@ -1134,6 +1135,7 @@ pub fn main() {
                 .long("expected-shred-version")
                 .value_name("VERSION")
                 .takes_value(true)
+                .validator(is_shred_version)
```
Comment (mvines) [inline validator/src/main.rs:1138]: I think this would work, less code:
```suggestion
                .validator(is_parsable::<u16>)
```

Comment (ryoqun) [reply validator/src/main.rs]: this pr got literally 1-line change: https://github.com/solana-labs/solana/pull/14183/commits/6094a360bbee86cb48076ae9c8100a54391ee1a5

Comment (mvines) [reply validator/src/main.rs]: perfect!

**Pull request merged**